src/sortStudents.ts

```diff
@@ -1,16 +1,71 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverageGrade(students: Student): number {
+  return (students.grades.reduce((sum, grade) => sum + grade))
+    / students.grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyStudents = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      copyStudents.sort((prev, curr) => {
+        return order === 'asc'
+          ? prev[sortBy].localeCompare(curr[sortBy])
+          : curr[sortBy].localeCompare(prev[sortBy]);
```

Comment:
You can do like this
```suggestion
  const copyStudents = [...students];
  const check = order === 'asc';
  
  switch (sortBy) {
    case SortType.Name:
    case SortType.Surname:
      copyStudents.sort((prev, curr) => {
        return check
          ? prev[sortBy].localeCompare(curr[sortBy])
          : curr[sortBy].localeCompare(prev[sortBy]);
```